packages/plugable/src/plugable.ts

```diff
@@ -0,0 +1,96 @@
+import { type Plugable, type Key, internals, PlugableInternals, Val } from './types';
+
+const proto = {
```

Comment:
maybe inline the functions on the proto instead of proxy them?

also set the `Plugable` type.